src/vm/class.cpp

```diff
@@ -3195,3 +3195,706 @@ void EEClass::SetPackableField(EEClassFieldId eField, DWORD dwValue)
     _ASSERTE(!m_fFieldsArePacked);
     GetPackedFields()->SetUnpackedField(eField, dwValue);
 }
+
+#ifndef DACCESS_COMPILE
+
+//=======================================================================
+// Called from the clsloader to load up and summarize the field metadata
+// for layout classes.
+//
+// Warning: This function can load other classes (esp. for nested structs.)
+//=======================================================================
+#ifdef _PREFAST_
+#pragma warning(push)
+#pragma warning(disable:21000) // Suppress PREFast warning about overly large function
+#endif
+VOID EEClassLayoutInfo::CollectLayoutFieldMetadataThrowing(
```

Comment:
The perf overhead of just this call is probably amazing - so many arguments.

Reply:
Yeah and it's hit on type load for every type with non-auto layout. And `ParseNativeType` (with 11 arguments in release, 14 in debug) gets hit once for every instance field of every loaded type with non-auto layout.